sklearn/ensemble/forest.py

```diff
@@ -597,6 +597,64 @@ def predict_log_proba(self, X):
 
             return proba
 
+    def predict_uplift(self, X):
+        """Predict uplift probabilities for X.
+
+        uplift probabilities represent the
+
+        [P(ACTIVE|CONTROL),P(NON-ACTIVE|CONTROL),P(ACTIVE|TREAT),P(NON-ACTIVE|TREAT)]
+
+        The predicted uplift probabilities of an input sample is computed as
+        the mean predicted uplift probabilities of the trees in the forest. The
+        uplift probability of a single tree is the fraction of samples of the same
+        class in a leaf.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix of shape = [n_samples, n_features]
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        p : array of shape = [n_samples, n_classes], or a list of n_outputs
+            such arrays if n_outputs > 1.
+            The class probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute `classes_`.
+        """
+        # Check data
+        X = self._validate_X_predict(X)
+
+        # Assign chunk of trees to jobs
+        n_jobs, _, _ = _partition_estimators(self.n_estimators, self.n_jobs)
+
+        # Parallel loop
```

Comment:
Can you fix these comments?  No new line between comments and the block below.

Same with "# Reduce" below.